rust/src/server.rs

```diff
@@ -259,6 +261,15 @@ fn initialize(
     );
     state.phase = IndexerPhase::Watching;
 
+    // flush/compress database
+    let random_number = rand::thread_rng().gen_range(1..=100);
```

Comment:
Suggest avoiding the dependency on 'rand' by simply choosing a suffix ('tmp-checkpoint') and appending it. It is not likely to last a long time on the filesystem, so little chance of conflict with another temporary snapshot.